src/redirects-manager.tsx

```diff
@@ -1,238 +1,60 @@
-            } catch (e) {
-                this.showNotification('error', 'An Error ocurred! Changes not saved');
-                return e;
-            }
-            if (result.ok) {
-                let json: {data : {redirects_added: number, redirects_modified: number, redirects_deleted: number, store: Array<WerRedirectionData>}};
-                try {
-                    json = await result.json();
-                } catch (e) {
-                    this.showNotification('error', 'An Error ocurred! Changes not saved');
-                    return e;
-                }
-                if (json.data.redirects_added === 0 && json.data.redirects_modified === 0 && json.data.redirects_deleted === 0)
-                {
-                    this.showNotification('warning', 'No changes were made!');
-                } else {
-                    this.showNotification('success', ({msg}) => { 
-                        return (
-                        <div>Redirects Succesfully saved!<br/>
-                        Added: {json.data.redirects_added}<br/>
-                        Modified: {json.data.redirects_modified}<br/>
-                        Deleted: {json.data.redirects_deleted}</div>
-                        )
-                    });
-                }
-                let newState = this.state;
-                newState.store = this.validateStore(json.data.store);
-                newState.saving = false;
-                newState.lastSave = result.ok;
-                this.setState(newState);
-            } else {
-                this.showNotification('error', 'An Error ocurred! Changes not saved');
-            }
-            return;
-        }
+export const RedirectsManager = ({initialState}: RedirectsManagerProps) => {
+    const [state, dispatch] = React.useReducer(redirectsManagerReducer, initialState);
+    return (
+        <div>
+            <table className="widefat" style={{width: '100%'}}>
+                <Toolbar dispatch={dispatch} state={state} />
+                <thead>
+                    <tr>
+                        <th style={{width: '41%'}} >Request</th>
+                        <th style={{width: '2%'}} ></th>
+                        <th style={{width: '41%'}} >Destination</th>
+                        <th style={{width: '11%'}} >Last Modification</th>
+                        <th style={{width: '5%', textAlign: 'center'}} >Action</th>
+                    </tr>
+                </thead>
+                <ListRedirections
+                    store={state.store}
+                    filterBy={state.filterBy}
+                    perPage={state.perPage}
+                    currentPage={state.currentPage}
+                    dispatch={dispatch} />
+            </table>
+            <ToastContainer position="bottom-right"/>
+        </div>
+    );
+};
+RedirectsManager.displayName = 'RedirectsManager';
+
+export class MountRedirectsManager extends React.Component<RedirectsManagerProps> {
```

Comment:
Do we need this component? Why not use `RedirectsManager` directly?
Regarding `defaultProps` - you can set them for FCs as well:
```
RedirectsManager.defaultProps: { ... }
```

Reply:
I had trouble with `mountComponent` working directly with the FC, and thats why I leave the mounter there, but let me give it a try with  `.defaulProps:`

Reply:
This is the only thing missing. I had not found a way to mount the FC directly with our package